### PR TITLE
Fix axis labeling issue when flipped

### PR DESCRIPTION
**Pull request by Kusefiru**
Related to issue #2019

I've also removed some commented code, I don't know if I should have.

### CODE REVIEW

File: vispy/visuals/axis.py [hunk 1]
```diff
@@ -412,12 +410,15 @@ def _get_tick_frac_labels(self):
...
-            major_frac = major_frac[::-1] if flip else major_frac
             use_mask = (major_frac > -0.0001) & (major_frac < 1.0001)
             major_frac = major_frac[use_mask]
             labels = [l for li, l in enumerate(labels) if use_mask[li]]
             minor_frac = minor_frac[(minor_frac > -0.0001) &
                                     (minor_frac < 1.0001)]
+            # Flip ticks coordinates if necessary :
+            if flip:
+                major_frac = 1-major_frac
+                minor_frac = 1-minor_frac
```
Comment (djhoese) [inline vispy/visuals/axis.py:421]:
```suggestion
                major_frac = 1 - major_frac
                minor_frac = 1 - minor_frac
```

**Review by djhoese — CHANGES_REQUESTED** [review]
Thanks for doing this fix as a smaller pull request. Much easier to review. I've made a small style suggestion inline. Otherwise, do you think you could add a test to `vispy/visuals/tests/test_axis.py`? I'm not sure if there is a property that would be easy to check or some other easy way to add a test. From what I can tell the `flip` is never actually tested in the existing tests.

**Comment by Kusefiru** [comment]
> Otherwise, do you think you could add a test to vispy/visuals/tests/test_axis.py? I'm not sure if there is a property that would be easy to check or some other easy way to add a test. From what I can tell the flip is never actually tested in the existing tests.

I'm not really sure about how this could be tested. Maybe from `self._ticks.pos` ?

I can give it a look, if you are okay with waiting a bit.

**Comment by djhoese** [comment]
Yeah, no rush. It'd be nice to include this fix in 0.7, but I have a lot of work to do on another PR before that's ready.

**Comment by djhoese** [comment]
@Kusefiru What's your availability to add tests for this?

**Comment by Kusefiru** [comment]
I added two tests, tell me if they are enough or if I missed anything.

File: vispy/visuals/tests/test_axis.py [hunk 2]
```diff
@@ -64,4 +64,57 @@ def test_rotation_angle():
...
+
+    canvas = scene.SceneCanvas(keys=None, size=(800, 600), show=True)
+    view = canvas.central_widget.add_view()
+    view.camera = scene.cameras.PanZoomCamera(parent=view.scene)
+
+    axis1 = visuals.Axis(pos=[[-1.0, 0], [1.0, 0]], domain=(0., 1.25), parent=view.scene)
+
+    canvas.draw_visual(axis1)
+    # Get a nice array of x ticks positions
+    x_ticks_positions = axis1._ticks.pos[:, ::2][::2].flatten()
```
Comment (djhoese) [inline vispy/visuals/tests/test_axis.py:106]: Why do you take every other (`::2`) point?

Comment (Kusefiru) [reply vispy/visuals/tests/test_axis.py:106]: Ticks positions are given for each tick line, so each tick has two (x, y) positions. As the axis is x-oriented, for each line, both x positions are equals so we can remove half of them.

As we are not interested in y positions either (again, we are testing an x-oriented axis), we can again remove half of the points.

Comment (djhoese) [reply vispy/visuals/tests/test_axis.py:106]: Got it. Sounds good. Seeing as your comment is from 8 hours ago, do you plan on making that commit with the change?

**Review by djhoese — APPROVED** [review]

**Comment by Kusefiru** [comment]
Changes done, the request can be accepted if there's nothing more to add.

**Comment by djhoese** [comment]
You rock! Glad such a basic feature is now fixed. Thank you.

**Pull request merged**